packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -1121,5 +1127,491 @@ describe('KeyringController', () => {
+        version: 'mockVersion',
+      };
+
+      const mockedAppVersionValueBolos = {
+        appName: 'BOLOS',
+        version: 'mockVersion',
+      };
+
+      getAppAndVersionStub.onCall(0).resolves(mockedAppVersionValue);
+      getAppAndVersionStub.onCall(1).resolves(mockedAppVersionValueBolos);
+
+      const quitAppSpy = sinon.stub(ledgerKeyring, 'quitApp');
+      quitAppSpy.resolves();
+
+      // Testing the path when app name is not BOLOS
+      await keyringController.closeRunningAppOnLedger();
+      expect(quitAppSpy.callCount).toBe(1);
+
+      // Reset the quitAppSpy call count
+      quitAppSpy.reset();
+
+      // Testing the path when the app name is BOLOS
+      await keyringController.closeRunningAppOnLedger();
+      expect(quitAppSpy.callCount).toBe(0);
+    });
+
+    it('should update the state when unlocking the default account on ledger', async () => {
+      // Setup
+
+      const updateIdentitesSpy = sinon.spy(preferences, 'updateIdentities');
+      const setAccountLabelSpy = sinon.spy(preferences, 'setAccountLabel');
+      const setSelectedAddressSpy = sinon.spy(
+        preferences,
+        'setSelectedAddress',
+      );
+
+      const persistAllKeyringsSpy = sinon.spy(
+        EthKeyringController.prototype,
+        'persistAllKeyrings',
+      );
+
+      // creating a new keyring controller without adding a ledger account
+      const locallyUsedKeyring = new KeyringController(
+        {
+          setAccountLabel: preferences.setAccountLabel.bind(preferences),
+          removeIdentity: preferences.removeIdentity.bind(preferences),
+          syncIdentities: preferences.syncIdentities.bind(preferences),
+          updateIdentities: preferences.updateIdentities.bind(preferences),
+          setSelectedAddress: preferences.setSelectedAddress.bind(preferences),
+        },
+        baseConfig,
+      );
+
+      await locallyUsedKeyring.createNewVaultAndKeychain(password);
+      const localLedgerKeyring = await locallyUsedKeyring.getLedgerKeyring();
+
+      localLedgerKeyring.setApp({
+        signTransaction: sinon.stub(),
+        getAddress: sinon.stub().resolves({
+          address: '0xe908e4378431418759b4f87b4bf7966e8aaa5cf2',
+        }),
+        signEIP712HashedMessage: sinon.stub(),
+        signPersonalMessage: sinon.stub(),
+      });
+
+      const fullUpdateSpy = sinon.spy(locallyUsedKeyring, 'fullUpdate');
+
+      // Start of test
+      await locallyUsedKeyring.fullUpdate();
+
+      expect(fullUpdateSpy.callCount).toBe(1);
+
+      const { keyrings: beforeAddingLedger } = await fullUpdateSpy
+        .returnValues[0];
+
+      // we check for the accounts in the index 1 position because 0 is the Hd Key tree.
+      expect(beforeAddingLedger[1].accounts).toStrictEqual([]);
+
+      // now adding the ledger account
+      const account = await locallyUsedKeyring.unlockLedgerDefaultAccount();
+
+      const updatedKeyrings = (await fullUpdateSpy.returnValues[1]).keyrings;
```

Comment:
It seems a bit strange to test the result of another method that's not under test here. Is it possible to test the state instead?

Reply:
just changed to use `getAccounts`